db/common/src/main/java/org/commonjava/indy/db/common/AbstractStoreDataManager.java

```diff
@@ -294,14 +294,17 @@ protected boolean store( final ArtifactStore store, final ChangeSummary summary,
             ArtifactStoreValidateData validateData = null ;
             try {
 
-                if(internalFeatureConfig.getSslValidation() ) {
-                    validateData = storeValidator.validate(store);
+
+                if(configuration.isSSLRequired()) {
+                    if (internalFeatureConfig.getSslValidation()) {
```

Comment:
getSslValidation() is really intended to be getValidationEnabled() or isValidationEnabled()...which means it should control the functioning of this whole validator. That was correct before this PR.

The new configuration.isSSLRequired() check should be called in DefaultStoreValidator, where it will determine whether the SSL requirement is enforced or if we just do basic HTTP availability checks. If we disable the whole validation feature using InternalFeatureConfig, then neither of those will run.

Reply:
But then problem with that setup would be remote repositories like this last one - ( npm repo with this problem ) .
Example and real case: 
This npm remote repo was SSL and available for HTTP requests but because it was behind secure access with required certificate our HTTP Client was getting  not allow entry Exceptions :
================================================================
Not Successfull HTTP GET request from StoreValidatorRemote, 
 => Exception: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
14:53:08.864 [store-validation-0] ERROR o.c.indy.data.DefaultStoreValidator -  => Not Successfull HTTP HEAD request from StoreValidatorRemote, 
 => Exception: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
===================================================================

Maybe solution would be that we just return just information about validation of validated repository and not to disable them , like you mention yesterday , and later on just list that information like list of suspicious remote repositories?
The other solution is for us to provice certificate files in $JAVA_HOME/jre/lib/security/ for every SSL remote repo ....

Reply:
In this case, we have a HTTPS repository that the user specified. This is something that should *pass* our validation rule for SSL repository URLs. However, it will still fail the HTTP HEAD/GET/OPTIONS/whatever validation we might want to do, and would be marked as disabled. I'd argue that your validator is doing exactly the right thing.

We have two options for handling this, both of which are IMO valid use cases (not bugs): We can change the system-wide SSL CAs that we use to include the one that issued this certificate. Or, we can configure the server cert directly on the remote repository definition. We have per-repository SSL certificate support for exactly this reason...to prevent situations where users cannot add repositories simply because they can't redeploy the system, and the server cert isn't in the system's normal CA list.